dbms/src/Storages/DeltaMerge/BitmapFilter/BitmapFilter.cpp

```diff
@@ -0,0 +1,130 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Storages/DeltaMerge/BitmapFilter/BitmapFilter.h>
+#include <Storages/DeltaMerge/DeltaMergeHelpers.h>
+#include <Storages/DeltaMerge/Segment.h>
+
+namespace DB::DM
+{
+BitmapFilter::BitmapFilter(UInt32 size_, const SegmentSnapshotPtr & snapshot_, bool default_value)
+    : filter(size_, default_value)
+    , snap(snapshot_)
+    , all_match(default_value)
+{}
+
+void BitmapFilter::set(BlockInputStreamPtr & stream)
+{
+    stream->readPrefix();
+    for (;;)
+    {
+        FilterPtr f = nullptr;
+        auto blk = stream->read(f, /*res_filter*/ true);
+        if (likely(blk))
+        {
+            set(blk.segmentRowIdCol(), f);
+        }
+        else
+        {
+            break;
+        }
+    }
+    stream->readSuffix();
+}
```

Comment:
Feel confused about this function. When the stream is `DMVersionFilterBlockInputStream`, the return FilterPtr is associated with the block return by `DMRowKeyFilterBlockInputStream` rather than the original block in dmfile, why it can be used for generating bitmap?

Reply:
get it